File: eventix-client/src/types/event.ts
export interface Event {
  id: string
  creator_id?: string
  title: string
  slug: string
  description: string
  start_time: string
  location_name?: string | null
  location_address?: string | null
  image_url?: string | null
  is_published: boolean
  created_at?: string
  updated_at?: string
}

export interface CreateEventRequest {
  title: string
  slug: string
  description: string
  start_time: string
  location_name?: string
  location_address?: string
  image_url?: string // Base64 encoded image
  is_published: boolean
  city_id: number
  category_ids: number[]
  ticket_types: TicketTypeRequest[]
}

export interface UpdateEventRequest {
  title?: string
  slug?: string
  description?: string
  start_time?: string
  location_name?: string | null
  location_address?: string | null
  image_url?: string // Base64 encoded image
  is_published?: boolean
}

export interface TicketTypeRequest {
  name: string
  description: string
  price_cents: number
  total_quantity: number
}

export interface TicketType {
  id: string
  name: string
  description: string
  price_cents: number
  total_quantity: number
  available_quantity: number
  created_at: string
  updated_at: string
}

export interface Ticket {
  id: string
  order_id: string
  ticket_type_id: string
  code: string
  status: boolean // true if used, false if not
  created_at: string
  user_email: string
  ticket_type_name: string
}

export interface Order {
  id: string
  user_id: string
  buyer_email: string
  status: string
  total_amount: number // in cents
  payment_ref: string
  created_at: string
}

File: EventTix-backend/internal/models/event.go
package models

import "time"

type Event struct {
	ID              string     `json:"id"` // UUID
	Title           string     `json:"title"`
	Slug            string     `json:"slug"`
	Description     string     `json:"description"`
	StartTime       time.Time  `json:"start_time"`
	LocationName    *string    `json:"location_name"`    // Pointer to allow NULL in DB
	LocationAddress *string    `json:"location_address"` // Pointer to allow NULL in DB
	ImageURL        *string    `json:"image_url"`        // Pointer to allow NULL in DB
	IsPublished     bool       `json:"is_published"`
	CreatedAt       time.Time  `json:"created_at"`
	UpdatedAt       *time.Time `json:"updated_at"` // Pointer to allow NULL in DB
}

// In internal/adminHandlers or internal/models
type UpdateEventRequest struct {
	Title           string               `json:"title"`
	Slug            string               `json:"slug"`
	Description     string               `json:"description"`
	StartTime       time.Time            `json:"start_time"`
	LocationName    *string              `json:"location_name"`    // Use pointer for optional/nullable fields
	LocationAddress *string              `json:"location_address"` // Use pointer for optional/nullable fields
	ImageURL        *string              `json:"image_url"`        // Use pointer for optional/nullable fields (can be nil, empty string, or base64)
	IsPublished     bool                 `json:"is_published"`
	CityID          int                  `json:"city_id"`
	CategoryIDs     []int                `json:"category_ids"`
	TicketTypes     []TicketTypeUpdateIn `json:"ticket_types"` // Reference the struct below
} // In internal/models (recommended)
type TicketTypeUpdateIn struct {
	ID            *int    `json:"id,omitempty"` // Pointer + omitempty means it's optional in JSON
	Name          string  `json:"name"`
	Description   *string `json:"description"` // Description can be optional/nullable
	PriceCents    int     `json:"price_cents"`
	TotalQuantity int     `json:"total_quantity"`
	// Note: AvailableQuantity is calculated on the server side, not sent by the client
}
